### PR TITLE
Add grafana data persistence support.

### LINKED ISSUE #154: Grafana data lost once pod restart.
**Is your feature request related to a problem? Please describe.**
Now, pulsar-helm-chart use empty dir instead of pv. So once grafana pod restarts, the grafana data lost.

**Describe the solution you'd like**
Add grafana data persistence support.

### PR DESCRIPTION
**Pull request by dragonls**
Fixes #154 

### Motivation

Add pulsar pvc to support grafana data persistence.

### Modifications
- Add `grafana-pvc.yaml`.
- Add grafana data mount logic in `grafana-deployment.yaml`
- Add `grafana.volumns` option in `values.yaml` to control the grafana data volumes.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### CODE REVIEW

File: charts/pulsar/templates/grafana-pvc.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  namespace: {{ template "pulsar.namespace" . }}
+spec:
+  resources:
+    requests:
+      storage: {{ .Values.grafana.volumes.data.size }}
+  accessModes: [ "ReadWriteOnce" ]
+{{- if .Values.grafana.volumes.data.storageClassName }}
+  storageClassName: "{{ .Values.grafana.volumes.data.storageClassName }}"
+{{- else if and (not (and .Values.volumes.local_storage .Values.grafana.volumes.data.local_storage)) .Values.grafana.volumes.data.storageClass }}
+  storageClassName: "{{ template "pulsar.fullname" . }}-{{ .Values.grafana.component }}-{{ .Values.grafana.volumes.data.name }}"
```
Comment (MarvinCai) [inline charts/pulsar/templates/grafana-pvc.yaml:35]: other components create their own storageclass so we can use this storageclass, seem that's not true with grafana here, then we should either remove this or add creation for storageclass.

Comment (dragonls) [reply charts/pulsar/templates/grafana-pvc.yaml:35]: Add creation for grafana storageclass.

File: charts/pulsar/templates/grafana-deployment.yaml [hunk 2]
```diff
@@ -88,4 +91,20 @@ spec:
               name: "{{ template "pulsar.fullname" . }}-{{ .Values.grafana.component }}-secret"
               key: GRAFANA_ADMIN_PASSWORD
         {{- include "pulsar.imagePullSecrets" . | nindent 6}}
+      volumes:
+        {{- if not (and (and .Values.persistence .Values.volumes.persistence) .Values.grafana.volumes.persistence) }}
```
[the remark this answers is not in the document]
Comment (dragonls) [reply charts/pulsar/templates/grafana-deployment.yaml]: Simplified.

**Comment by dragonls** [comment]
Sorry for the late modification. @MarvinCai PTAL.

File: charts/pulsar/values.yaml [hunk 3]
```diff
@@ -988,6 +988,24 @@ grafana:
     requests:
       memory: 250Mi
       cpu: 0.1
+  volumes:
+    # use a persistent volume or emptyDir
+    persistence: true
+    data:
+      name: data
+      size: 10Gi
```
Comment (lhotari) [inline charts/pulsar/values.yaml:996]: 10Gi seems a lot for grafana persistence. I'd assume that 100Mi would be sufficient.

File: charts/pulsar/values.yaml [hunk 4]
```diff
@@ -988,6 +988,24 @@ grafana:
     requests:
       memory: 250Mi
       cpu: 0.1
+  volumes:
+    # use a persistent volume or emptyDir
+    persistence: true
```
Comment (michaeljmarshall) [inline charts/pulsar/values.yaml:993]: What is the reasoning for making this the default behavior? Alternatively, we could mount dashboards as config maps and avoid the extra storage costs associated with a PVC.

Comment (dragonls) [reply charts/pulsar/values.yaml:993]: Just thinking that user might be confused if the grafana pod restarts and data lost, once user modified the grafana data such as the dashboards.

File: charts/pulsar/values.yaml (hunk 3 above)
Comment (dragonls) [reply charts/pulsar/values.yaml]: Updated.

File: charts/pulsar/values.yaml (hunk 4 above)
Comment (michaeljmarshall) [reply charts/pulsar/values.yaml:993]: I agree there is a chance for confusion. However, the official Grafana helm chart disables persistence by default, so I think we should too.

https://github.com/grafana/helm-charts/blob/2a422f6c43c3db33d31fc12d23957d5b6ff5e8c6/charts/grafana/values.yaml#L275-L277

Comment (dragonls) [reply charts/pulsar/values.yaml]: I agree, change default persistence value to false.

**Pull request closed without merging**